### DISCUSSION TITLE
Unable to change name when using authentication provider

### DISCUSSION THREAD

**Discussion by jonpetersathan** [requarks/wiki#5109] [Error / Bug Report] [answered]
The display name always defaults back to the name given by the authentication provider. I tested this with Google as auth provider. The option to change the display name still appears under profile settings, however after logging out and logging in again, the name will be reset to whatever name is given by the auth provider.

**Comment by jonpetersathan** [comment]
The simplest solution would be to remove this line: https://github.com/Requarks/wiki/blob/2493fd85ae46573e3c2df675d04c7591159f0e8b/server/models/users.js#L239

However, if this is the expected behaviour, maybe it makes sense to introduce a custom flag to track if the user manually changed the display name?

**Reply by Nuuki9** [reply]
I'd also like to see this addressed if possible. It clearly makes sense to pull in the display name from the external service initially, but if its subsequently manually changed in wiki.js it shouldn't get reset back at next login.

**Comment by NGPixel** [comment, marked as answer]
https://feedback.js.wiki/wiki/p/for-external-identity-providers-separate-display-name-from-username
